src/pool/srv_pool.c

```diff
@@ -3775,6 +3775,248 @@ pool_find_all_targets_by_addr(uuid_t pool_uuid,
+	}
+
+	/* Copy the data */
+	uuid_copy(arg->next->pic_hdl, *uuid);
+	arg->next->pic_flags = hdl->ph_flags;
+	arg->next->pic_capas = hdl->ph_sec_capas;
+	arg->next->pic_cred_size = lookup_hdl->sph_cred.iov_buf_len;
+	memcpy(arg->next->pic_creds, lookup_hdl->sph_cred.iov_buf,
+	       lookup_hdl->sph_cred.iov_buf_len);
+
+	/* Adjust the pointers for the next iteration */
+	arg->hdls_used = size_needed;
+	arg->next = (struct pool_iv_conn *)
+		(((char *)*arg->hdls) + arg->hdls_used);
+
+out_hdl:
+	ds_pool_hdl_put(lookup_hdl);
+
+	return DER_SUCCESS;
+}
+
+/**
+ * Retrieves a flat buffer containing all currently open handles
+ *
+ * \param pool_uuid [IN]  The pool to get handles for
+ * \param hdls      [OUT] A flat-packed buffer of all open handles
+ *                        (struct pool_iv_conn). Caller must free hdls->iov_buf.
+ *                        Note that these are variable size, and can not be
+ *                        indexed like an array
+ * \param out_size  [OUT] The size of the buffer pointed to by hdls
+ *
+ * \return If no handles are currently open this will return DER_SUCCESS with
+ *         hdls = NULL and out_size = 0.
+ *         Otherwise returns DER_SUCCESS or an -error code
+ */
+int
+ds_pool_get_open_handles(uuid_t pool_uuid, d_iov_t *hdls)
+{
+	struct pool_svc			*svc;
+	struct redist_open_hdls_arg	 arg;
+	uint32_t			 connectable;
+	struct rdb_tx			 tx;
+	d_iov_t				 value;
+	uint32_t			 nhandles;
+	int				 rc;
+
+	d_iov_set(hdls, NULL, 0);
+
+	rc = pool_svc_lookup_leader(pool_uuid, &svc, NULL /* hint */);
+	if (rc != 0)
+		return rc;
+
+	rc = rdb_tx_begin(svc->ps_rsvc.s_db, svc->ps_rsvc.s_term, &tx);
+	if (rc != 0)
+		D_GOTO(out_svc, rc);
+
+	ABT_rwlock_rdlock(svc->ps_lock);
+
+	/* Check if pool is being destroyed and not accepting connections */
+	d_iov_set(&value, &connectable, sizeof(connectable));
+	rc = rdb_tx_lookup(&tx, &svc->ps_root,
+			   &ds_pool_prop_connectable, &value);
+	if (rc != 0)
+		D_GOTO(out_lock, rc);
+	if (!connectable) {
+		D_ERROR(DF_UUID": being destroyed, not accepting connections\n",
+			DP_UUID(pool_uuid));
+		D_GOTO(out_lock, rc = -DER_BUSY);
+	}
+
+	/* Check how many handles are currently open */
+	d_iov_set(&value, &nhandles, sizeof(nhandles));
+	rc = rdb_tx_lookup(&tx, &svc->ps_root, &ds_pool_prop_nhandles, &value);
+	if (rc != 0)
+		D_GOTO(out_lock, rc);
+
+	/* Abort early if there are no open handles */
+	if (nhandles == 0)
+		D_GOTO(out_lock, rc);
+
+	/* Preallocate an approximate amount of space for the handles and the
+	 * variable-size creds field which is not accounted for in the size of
+	 * the base structure. The goal here isn't to be exactly right - can't
+	 * know at this point how much space is actually needed. Ballparking
+	 * close enough just reduces the number of reallocations needed during
+	 * iteration
+	 */
+	D_ALLOC(hdls->iov_buf, nhandles * (sizeof(struct pool_iv_conn) + 160));
+	if (hdls->iov_buf == NULL)
+		D_GOTO(out_lock, rc=-DER_NOMEM);
```

Comment:
(style)  spaces required around that '=' (ctx:VxO)
(style)  space required before that '-' (ctx:OxV)